### ISSUE TITLE
eig_simple_rump!(...; side=:left) segfaults inside libarblib

### ISSUE THREAD

**Issue by kalmarek** [kalmarek/Arblib.jl#64]
```julia
julia> A = [
               0.6873474041954415 0.7282180564881044 0.07360652513458521
               0.000835810121029068 0.9256166870757694 0.5363310989411239
               0.07387174694790022 0.4050436025621329 0.20226010388885896
           ];

julia> B = [
               0.8982563031334123 0.3029712969740874 0.8585014523679579
               0.7583002736998279 0.8854763478184455 0.3031103325817668
               0.2319572749472405 0.5769840251057949 0.5119507333628952
           ];

julia> M = MatT(A + B * im, prec = 64);

julia> λs1, _ = Arblib.eig_simple_rump(M, side = :right)
(Acb[[1.30153309871612730 +/- 2.32e-18] + [1.80765893841642298 +/- 6.06e-18]*I, [0.682258049701900519 +/- 4.76e-19] + [0.394077524731353095 +/- 5.11e-19]*I, [-0.168566953257957990 +/- 5.20e-19] + [0.093946921166976970 +/- 5.44e-19]*I], Acb[[0.4449264898696659941 +/- 5.65e-20] + [0.4634294069416568042 +/- 6.91e-20]*I [-0.6798370500753717779 +/- 1.38e-20] - [0.2620245033973304097 +/- 2.86e-20]*I [0.631991508186129629 +/- 3.22e-19] + [-0.003670352145081746 +/- 3.99e-19]*I; [0.3906072184390861097 +/- 1.14e-20] + [0.5459402715947035059 +/- 3.27e-20]*I [0.4639872344939433396 +/- 3.81e-20] + [0.3039649915535414256 +/- 7.11e-20]*I [-0.210579599428613159 +/- 4.58e-19] + [-0.587107155572761272 +/- 3.24e-19]*I; [0.1739472525947631981 +/- 4.43e-20] + [0.3261766047144971842 +/- 4.30e-20]*I [-0.05774479193929384316 +/- 7.28e-21] + [0.4312026289542442071 +/- 6.97e-20]*I [-0.07234894806558699552 +/- 8.31e-22] + [0.6558941721622767361 +/- 2.06e-20]*I])

julia> λs2, _ = Arblib.eig_simple_rump(M, side=:left)

signal (11): Segmentation fault
in expression starting at REPL[338]:1
acb_mat_solve at /workspace/srcdir/arb-2.18.1/acb_mat/solve.c:17
Allocations: 63278843 (Pool: 63255393; Big: 23450); GC: 71
[1]    468126 segmentation fault (core dumped)  julia --sysimage ~/.julia/sys_custom.so --project

```

**Comment by Joel-Dahne** [comment]
Trying to look into this I noticed some bugs in Arblib. Though the crash does seem to be coming from Arb...

The problem in Arblib is that the call to `eigvals_approx, R_eigvecs_approx = approx_eig_qr(A)` does not use the `side` argument nor the `prec` argument. For example instead of 
```julia
        function $jlf(
            eigvals::AcbVectorLike,
            eigvecs::AcbMatrixLike,
            A::Union{AcbMatrix,AcbRefMatrix};
            prec = precision(A),
            side = :right,
        )
            eigvals_approx, R_eigvecs_approx = approx_eig_qr(A)
            return $jlf(
                eigvals,
                eigvecs,
                A,
                eigvals_approx,
                R_eigvecs_approx,
                prec = prec,
                side = side,
            )
        end
```
We should probably have
```julia
        function $jlf(
            eigvals::AcbVectorLike,
            eigvecs::AcbMatrixLike,
            A::Union{AcbMatrix,AcbRefMatrix};
            prec = precision(A),
            side = :right,
        )
            eigvals_approx, R_eigvecs_approx = approx_eig_qr(A, prec = prec, side = side)
            return $jlf(
                eigvals,
                eigvecs,
                A,
                eigvals_approx,
                R_eigvecs_approx,
                prec = prec,
                side = side,
            )
        end
```

With this change I don't think the crash occurs anymore. But giving the wrong matrix to Arb should not make it crash... I'll look into it more!

**Comment by Joel-Dahne** [comment]
I realised that `R_eigvecs_approx` should always be the right eigenvectors, so the side should not be given to `approx_eig_qr`. Though the precision should probably still be given.

**Comment by Joel-Dahne** [comment]
It's indeed a bug in Arb, it can be reproduced with
```C
#include "acb.h"
#include "acb_mat.h"

int
main()
{
  slong prec = 64;
  mag_t tol;
  acb_ptr E, E_approx;
  acb_mat_t L, R, A, L_approx, R_approx;

  mag_init(tol);

  E = _acb_vec_init(3);
  E_approx = _acb_vec_init(3);

  acb_mat_init(L, 3, 3);
  acb_mat_init(R, 3, 3);
  acb_mat_init(A, 3, 3);
  acb_mat_init(L_approx, 3, 3);
  acb_mat_init(R_approx, 3, 3);

  // Setup matrix
  acb_set_d_d(acb_mat_entry(A, 0, 0), 0.6873474041954415, 0.8982563031334123);
  acb_set_d_d(acb_mat_entry(A, 0, 1), 0.7282180564881044, 0.3029712969740874);
  acb_set_d_d(acb_mat_entry(A, 0, 2), 0.07360652513458521, 0.8585014523679579);
  acb_set_d_d(acb_mat_entry(A, 1, 0), 0.000835810121029068, 0.7583002736998279);
  acb_set_d_d(acb_mat_entry(A, 1, 1), 0.9256166870757694, 0.8854763478184455);
  acb_set_d_d(acb_mat_entry(A, 1, 2), 0.5363310989411239, 0.3031103325817668);
  acb_set_d_d(acb_mat_entry(A, 2, 0), 0.07387174694790022, 0.2319572749472405);
  acb_set_d_d(acb_mat_entry(A, 2, 1), 0.4050436025621329, 0.5769840251057949);
  acb_set_d_d(acb_mat_entry(A, 2, 2), 0.20226010388885896, 0.5119507333628952);

  acb_mat_printd(A, 10);

  acb_mat_approx_eig_qr(E_approx, L_approx, R_approx, A, NULL, 0, prec);
  for (int i = 0; i < 3; i++) {
    acb_printd(E_approx + i, 10);
    flint_printf("\n");
  }
  acb_mat_printd(L_approx, 10);

  /* This one works with R equal to NULL */
  acb_mat_eig_simple_vdhoeven_mourrain(E, L, NULL, A, E_approx, R_approx, prec);
  flint_printf("vdhoeven_mourrain worked\n");
  /* This one works when R us not NULL*/
  acb_mat_eig_simple_rump(E, L, R, A, E_approx, R_approx, prec);
  flint_printf("rump worked with R != NULL\n");
  /* But crashes when R is NULL*/
  acb_mat_eig_simple_rump(E, L, NULL, A, E_approx, R_approx, prec);
  flint_printf("rump worked with R == NULL\n");
}
```
The problem is this part in Arb
```C
    if (L != NULL)
    {
        if (!result || !acb_mat_inv(L, R, prec))
            acb_mat_indeterminate(L);
    }
```
which does not check if `R` is non-null. I can report it to Arb, though the fix will probably be to require `R` to be non-null in case you want `L` or allocate it in the function. So we can work around it by constructing a temporary matrix.

**Comment by Joel-Dahne** [comment]
I opened an [issue](https://github.com/fredrik-johansson/arb/issues/329) in Arb.

**Comment by kalmarek** [comment]
yeah, precision should be passed to `approx_eig_qr`; Many thanks for detective work!

**Comment by Joel-Dahne** [comment]
The bug is fixed in Arb now as well!

**Comment by kalmarek** [comment]
ok, let's wait for the new arblib_jll

**Comment by kalmarek** [comment]
precision is passed to `approx_eig_qr` as of #68

**Comment by Joel-Dahne** [comment]
Closed by #111
